### PR TITLE
Rename Button Component's `Style` Prop to `Variant`

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
http://vjira2:8080/browse/BDS-1104

## Summary
Globally deprecates and renames the old `style` and `color` props to use the new `variant` prop name.

## Reference Examples 
- http://designsystem.morningstar.com/components/buttons.html#variations
- https://comet.discoveryeducation.com/components/buttons.html
- https://developer.salesforce.com/docs/component-library/bundle/lightning:button/example
- https://react-bootstrap.github.io/components/buttons/

## How to test
- Confirm Jest tests passing

## TODOs
- [x] Rename original `variation` prop name to `variant` per feedback and discussion from @mikemai2awesome 
- [ ] We should also add some button-specific tests to confirm that old `color` AND `style` props will continue to work as expected on the button component (both via Twig + Web Component), yes?

### CODE REVIEW

**Comment by mikemai2awesome** [comment]
@sghoweri 

My only concern is that `variation` (or style, color) is not enforcing the purpose behind these styles, which are supposed to communicate information architecture. When using this prop, the user should get a sense of what the button's [visual hierarchy](https://www.interaction-design.org/literature/topics/visual-hierarchy) is within the page layout. Furthermore, `primary`, `secondary`, and `text` do not belong in the same prop, `text` is a style while `primary` and `secondary` are hierarchy. 

How about something like the following?

```
hierarchy:
  enum:
    - primary
    - secondary
    - danger
variant:
  enum:
    - basic
    - plain
    - ghost (not in use yet)
```

**Comment by mikemai2awesome** [comment]
@sghoweri Any way we can make sure all the button changes/refactors are released together? This one (https://github.com/bolt-design-system/bolt/pull/1115) is from a while ago, also involves renaming/changing props.

File: packages/components/bolt-button/src/button.twig [hunk 1]
```diff
@@ -27,7 +34,7 @@
 
 {# Set up checks to validate that the component's prop values are allowed, based on the component's schema #}
 {% set size_options = schema.properties.size.enum %}
-{% set style_options = schema.properties.style.enum %}
+{% set variations = schema.properties.variant.enum %}
```
Comment (mikemai2awesome) [inline packages/components/bolt-button/src/button.twig]: Can we keep this consistent with other props? Use the `_options` postfix.

**Comment by sghoweri** [comment]
> @sghoweri Any way we can make sure all the button changes/refactors are released together? This one (#1115) is from a while ago, also involves renaming/changing props.

@mikemai2awesome I'd prefer we tackled each piece of this one at a time so we don't overlook anything, however I'd be fine batching the lot of refactor work to release it all at once.

**Comment by danielamorse** [comment]
@sghoweri do I have this right?
1. This is a v3.0 schema change
2. This work needs to be re-done because the code is so old

**Comment by sghoweri** [comment]
> @sghoweri do I have this right?
> 
@danielamorse see notes below:

> 1. This is a v3.0 schema change
Yes if this introduces a major breaking change, no if it isn't. 

And considering that I opened this PR almost a year ago... at this point, I'd recommend we operate as if we'll hit a `v2.50` release before we hit `v3.0` -- meaning we should try to get tech debt and fixes in place sooner rather than later so we aren't putting things off indefinitely.

> 2. This work needs to be re-done because the code is so old
Yep, that's my recommendation.

**Comment by danielamorse** [comment]
Closing, work will be redone in new PR.

**Pull request closed without merging**